### PR TITLE
Add Zarr IO tutorial

### LINKED ISSUE #1832: [Documentation]: Need tutorial for writing in Zarr
### What would you like changed or added to the documentation and why?

I just did a search over the PyNWB RDT for "Zarr" and came up with nothing. This seems like it would be a common route for users, so we should provide some guidance.

### Do you have any interest in helping write or edit the documentation?

Yes.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you checked the [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this change was not already [requested](https://github.com/NeurodataWithoutBorders/pynwb/issues)?

### PR DESCRIPTION
**Pull request by bendichter**
## Motivation

fix #1832 

## Checklist

- [x] Did you update CHANGELOG.md with your changes?
- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR clearly describes the problem and the solution?
- [x] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

File: docs/gallery/advanced_io/zarr_io.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+"""
+Zarr IO
+=======
+
+Zarr is an alternative backend option for NWB files. It is a Python package that
+provides an implementation of chunked, compressed, N-dimensional arrays. Zarr is a good
+option for large datasets because, like HDF5, it is designed to store data on disk and
+only load the data into memory when needed. Zarr is also a good option for parallel
+computing because it supports concurrent reads and writes.
```
Comment (oruebel) [inline docs/gallery/advanced_io/zarr_io.py:9]: I would suggest adding a note here along the lines of: 1) We use the `Zarr.DirectoryStore` or `Zarr.NestedDrirectoryStore` 2) the storage formats are optimized for storage in cloud storage (e.g., S3), 3) for very large files, Zarr will create many files which can lead to issues for traditional file system (that are not cloud object stores) due to limitations on the number of files per directory (this affects local disk, GDrive, Dropbox etc.)

File: docs/gallery/advanced_io/zarr_io.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+# -----------------
+# To read NWB files from Zarr, replace the :py:class:`~pynwb.NWBHDF5IO` with the analogous
+# :py:class:`hdmf_zarr.nwb.NWBZarrIO`.
+
+with NWBZarrIO(path=absolute_path, mode="r") as io:
+    read_nwbfile = io.read()
+
+#######################################################################################
+# For more information, see the :hdmf-zarr:`hdmf-zarr documentation<>`.
+
```
Comment (oruebel) [inline docs/gallery/advanced_io/zarr_io.py]: I would make this a callout ``.. note::`` and also point to the NWB specific tutorials in addition to the general hdmf-zarr docs.

**Review by oruebel — COMMENTED** [review]
Thanks for adding this tutorial. I added a couple of minor suggestions, but overall this looks good to me.

**Review by oruebel — DISMISSED** [review]

File: docs/gallery/advanced_io/zarr_io.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+Zarr is an alternative backend option for NWB files. It is a Python package that
+provides an implementation of chunked, compressed, N-dimensional arrays. Zarr is a good
+option for large datasets because, like HDF5, it is designed to store data on disk and
+only load the data into memory when needed. Zarr is also a good option for parallel
+computing because it supports concurrent reads and writes.
+
+Note that the Zarr native storage formats are optimized for storage in cloud storage
+(e.g., S3). For very large files, Zarr will create many files which can lead to
+issues for traditional file system (that are not cloud object stores) due to limitations
```
Comment (rly) [inline docs/gallery/advanced_io/zarr_io.py]:
```suggestion
issues for traditional file systems (that are not cloud object stores) due to limitations
```

**Review by rly — DISMISSED** [review]

File: docs/gallery/advanced_io/zarr_io.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+Zarr is an alternative backend option for NWB files. It is a Python package that
+provides an implementation of chunked, compressed, N-dimensional arrays. Zarr is a good
+option for large datasets because, like HDF5, it is designed to store data on disk and
+only load the data into memory when needed. Zarr is also a good option for parallel
+computing because it supports concurrent reads and writes.
+
+Note that the Zarr native storage formats are optimized for storage in cloud storage
+(e.g., S3). For very large files, Zarr will create many files which can lead to
+issues for traditional file systems (that are not cloud object stores) due to limitations
+on the number of files per directory (this affects local disk, GDrive, Dropbox etc.).
```
Comment (rly) [inline docs/gallery/advanced_io/zarr_io.py]:
```suggestion
issues for traditional file systems (that are not cloud object stores) due to
limitations on the number of files per directory (this affects local disk,
GDrive, Dropbox etc.).
```

**Review by rly — DISMISSED** [review]

File: docs/gallery/advanced_io/zarr_io.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+only load the data into memory when needed. Zarr is also a good option for parallel
+computing because it supports concurrent reads and writes.
+
+Note that the Zarr native storage formats are optimized for storage in cloud storage
+(e.g., S3). For very large files, Zarr will create many files which can lead to
+issues for traditional file systems (that are not cloud object stores) due to
+limitations on the number of files per directory (this affects local disk,
+GDrive, Dropbox etc.).
+
+Zarr read and write is provided by the :hdmf-zarr:`hdmf-zarr` package. First, create an
```
Comment (rly) [inline docs/gallery/advanced_io/zarr_io.py]:
```suggestion
Zarr read and write is provided by the :hdmf-zarr:`hdmf-zarr package<>`. First, create an
```

**Review by rly — APPROVED** [review]

**Pull request merged**